ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/QueryStreamWriter.java

```diff
@@ -183,12 +185,13 @@ private void drain(final OutputStream out) throws IOException {
     }
   }
 
-  private class StreamsExceptionHandler implements Thread.UncaughtExceptionHandler {
+  private class StreamsExceptionHandler implements StreamsUncaughtExceptionHandler {
     @Override
-    public void uncaughtException(final Thread thread, final Throwable exception) {
-      streamsException = exception instanceof Exception
-          ? (Exception) exception
-          : new RuntimeException(exception);
+    public StreamThreadExceptionResponse handle(final Throwable throwable) {
+      streamsException = throwable instanceof Exception
+              ? (Exception) throwable
+              : new RuntimeException(throwable);
+      return StreamThreadExceptionResponse.SHUTDOWN_CLIENT;
```

Comment:
I couldn't find where this was retried so maybe just shutdown?

Reply:
Just for my own education: we have two classes `QueryStreamWriter` and `PushQueryPublisher` for push queries, one called by the old API `StreamedQueryResource` and the other called by the new `WSQueryEndpoint`. Why we need to maintain two callees?

Reply:
This is a good question, I would also like to know the answer

Reply:
We originally had an http1 chunked-encoding implementation for transient queries that used `QueryStreamWriter`. Then, to support C3 we added the websockets stuff, which used `PushQueryPublisher`. The new http2 stuff seems to have it's own way of doing this that isn't even handling errors (see `QueryEndpoint` and `BlockingQueryPublisher`). So tl;dr: tech-debt.

I think we should consider 2 follow-ups:
1. clean up the legacy `/query` endpoint - I don't think anything is even using this anymore
2. add error handling to the new http2 endpoints